### DISCUSSION TITLE
Changing the minElev parameter

### DISCUSSION THREAD

**Discussion by praveengi113** [semuconsulting/pyubx2#162] [Q&A] [answered]
Hi there,
I'm Praveen, I'm using the pyubx2 to interface with the UBLOX Neo M8N receiver.
I'm trying to change the elevation mask of the receiver by using the below message.

cfg_nav5 = UBXMessage(
        "CFG",  # Message class
        "CFG-NAV5",  # Message ID
        mask=0x0020,  # Mask indicating we're updating the minElev field
        dynModel=3,  # Dynamic Model (portable, can be adjusted if needed)
        fixMode=3,  # Fix mode (3D-only, adjust if necessary)
        minElev=mask  # Minimum Elevation Mask in degrees
    )

With this my message is getting acknowledged by the receiver. But the value of minElev is not getting changed.

**Comment by semuadmin** [comment, marked as answer]
Hi @praveengi113 

Can you please include your *complete* Python script and confirm which firmware version your M8N is running? Thanks

I can see there are a number of issues with the code snippet you've provided. I recommend having another look at the [pyubx2 README, specifically the section about message generation](https://github.com/semuconsulting/pyubx2/blob/master/README.md#generating) and the [message definition for the CFG-NAV5 message](https://www.u-blox.com/sites/default/files/products/documents/u-blox8-M8_ReceiverDescrProtSpec_UBX-13003221.pdf).

By way of comparison, here's a script that I've just successfully tested on an NEO-M8N receiver running firmware SPG 3.01 Protocol 18.00 (*this arbitrarily sets the minElev value to 65 degrees - the default value is 5 degrees*):

```python
from serial import Serial
from pyubx2 import UBXMessage, SET

cfg_nav5 = UBXMessage(
    "CFG",  # Message class
    "CFG-NAV5",  # Message ID
    SET,  # msgmode must be 'SET' (0x01)
    dyn=1,  # set dynamic model mask bit
    minEl=1,  # set minimum elevation mask bit
    posFixMode=0,
    drLim=0,
    posMask=0,
    timeMask=0,
    staticHoldMask=0,
    dgpsMask=0,
    cnoThreshold=0,
    utc=0,
    dynModel=3,  # Dynamic Model (portable, can be adjusted if needed)
    fixMode=3,  # Fix mode (3D-only, adjust if necessary)
    fixedAlt=0.0,
    fixedAltVar=1.0,
    minElev=65,  # required value of minimum elevation in degrees
    drLimit=0,
    pDop=25.0,
    tDop=25.0,
    pAcc=100,
    tAcc=350,
    staticHoldThresh=0,
    dgnssTimeOut=60,
    cnoThreshNumSVs=0,
    cnoThresh=0,
    staticHoldMaxDist=0,
    utcStandard=0,
)

with Serial("/dev/tty.usbmodem1201", 38400, timeout=3) as stream:
    stream.write(cfg_nav5.serialize())

```

Confirmation from PyGPSClient that settings have been applied:

<img width="332" alt="Screenshot 2024-10-13 at 09 08 00" src="https://github.com/user-attachments/assets/1c82cdf9-8608-47bc-96c6-3f5f6c3d55d0">

**Reply by praveengi113** [reply]
Hi @semuadmin 
Thank you for getting back to me.
I'm a noobie on this topic. I've given below my message which was not able to get the work done.
    
cfg_nav5 = UBXMessage(
        "CFG",  # Message class
        "CFG-NAV5",  # Message ID
        SET,
        dynModel=3,  # Dynamic Model (portable, can be adjusted if needed)
        fixMode=3,  # Fix mode (3D-only, adjust if necessary)
        minElev=mask,  # Minimum Elevation Mask in degrees
    )

Thanks to you, Now I got a bit of more understanding in generating the UBXMessages.
With your code, I've changed mine. and it worked.

My updated code:
    cfg_nav5 = UBXMessage(
        "CFG",  # Message class
        "CFG-NAV5",  # Message ID
        SET,
        dyn=1,  # set dynamic model mask bit
        minEl=1,  # set minimum elevation mask bit
        posFixMode=0,
        dynModel=3,  # Dynamic Model (portable, can be adjusted if needed)
        fixMode=3,  # Fix mode (3D-only, adjust if necessary)
        minElev=mask,  # Minimum Elevation Mask in degrees
    )

By setting dyn, minEl, posFixMode to 1, I was able to change the elevation mask.

Can you give me some guidance on recording raw ubx logs for GNSS processing?

**Reply by semuadmin** [reply]
> Thanks to you, Now I got a bit of more understanding in generating the UBXMessages. With your code, I've changed mine. and it worked.

Glad to hear you got it working.

> Can you give me some guidance on recording raw ubx logs for GNSS processing?

Depends on exactly what you mean by 'raw ubx logs' and 'GNSS processing'. If you simply want to record the binary UBX protocol output of a u-blox receiver, you could for example use the [gnssstreamer](https://github.com/semuconsulting/pygnssutils?tab=readme-ov-file#gnssstreamer) CLI utility which is installed with [pygnssutils](https://github.com/semuconsulting/pygnssutils), e.g.:

```shell
gnssstreamer --port /dev/ttyACM0 --baudrate 38400 --timeout 3 --format 2 --protfilter 2 --verbosity 2 --clioutput 1 --output ubxdata.bin
```

If, on the other hand, you mean 'raw' GNSS navigation and observation data (i.e. UBX RXM-SFRBX and/or RXM-RAWX messages) as a precursor to PPK processing, then you'll almost certainly need to capture the data as above but then convert it to so-called RINEX format using a utility like [RTKLIB](https://www.rtklib.com/). I'm unable to offer any assistance with this third party utility.
